state-fold/src/delegate_access/partition_events.rs

```diff
@@ -89,7 +89,7 @@ where
     P: PartitionProvider<E, D> + Send + Sync,
 {
     pub fn new(concurrent_workers: usize, provider: &'a P, partition_data: &'a D) -> Self {
-        let semaphore = Semaphore::new(concurrent_workers);
+        let semaphore = Semaphore::new(concurrent_workers + 1);
```

Comment:
Is this change intended? It doesn't seem related to the PR.

Reply:
This fixes the configuration `sf_concurrent_events_fetch`. If set to zero, the system silently hangs. Instead of requiring it to be >0, I decided to just add 1 where it's used. This was the documented behavior [here](https://github.com/cartesi/state-fold/blob/9ae672aa0bbaaeda0d07ea62e0e18b0eb9ef74c3/state-fold/src/env/environment.rs#L39).

I just added it to the commit message and changelog.

What do you think?